src/main/java/com/aws/greengrass/device/ClientDevicesAuthIPCAgent.java

```diff
@@ -0,0 +1,212 @@
+    private static final String CERT_TYPE_SERVER = "SERVER";
+    private static final String CERT_TYPE_CLIENT = "CLIENT";
+
+    @Inject
+    @Setter(AccessLevel.PACKAGE)
+    ClientDevicesAuthService clientDevicesAuthService;
+    @Inject
+    @Setter(AccessLevel.PACKAGE)
+    CertificateManager certificateManager;
+
+    public SubscribeToCertificateUpdatesOperationHandler subscribeToCertificateUpdatesOperationHandler(
+            OperationContinuationHandlerContext context) {
+        return new SubscribeToCertificateUpdatesOperationHandler(context);
+    }
+
+    class SubscribeToCertificateUpdatesOperationHandler
+            extends GeneratedAbstractSubscribeToCertificateUpdatesOperationHandler {
+        private final String serviceName;
+        private final Consumer<X509Certificate> serverCertificateCallback = this::forwardServerCertificates;
+        private final Consumer<X509Certificate[]> clientCertificateCallback = this::forwardClientCertificates;
+        private String csr;
+
+        protected SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context) {
+            super(context);
+            serviceName = context.getAuthenticationData().getIdentityLabel();
+        }
+
+        @Override
+        protected void onStreamClosed() {
+
+        }
+
+        @SuppressWarnings("PMD.PreserveStackTrace")
+        @Override
+        public SubscribeToCertificateUpdatesResponse handleRequest(
+                SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+            return translateExceptions(() -> {
+                try {
+                    clientDevicesAuthService.doAuthorization(serviceName);
+                } catch (AuthorizationException e) {
+                    throw new UnauthorizedError(e.getMessage());
+                }
+                byte[] publicKey = subscribeToCertificateUpdatesRequest.getEncryptionKeys().getPublicKey();
+                byte[] privateKey = subscribeToCertificateUpdatesRequest.getEncryptionKeys().getPrivateKey();
+                String algorithm =
+                        subscribeToCertificateUpdatesRequest.getEncryptionKeyOptions().getKeyAlgorithmAsString();
+                try {
+                    csr = getCSR(publicKey, privateKey, algorithm);
+                } catch (IOException | OperatorCreationException e) {
+                    logger.atError().log("Unable to create certificate signing request");
+                    throw new ServiceError(String.format("Subscribe to certificate update failed with error %s", e));
+                }
+                String certificateType = subscribeToCertificateUpdatesRequest.getCertificateType().getValue();
+                if (certificateType.equals(CERT_TYPE_SERVER)) {
+                    try {
+                        certificateManager.subscribeToServerCertificateUpdates(csr, serverCertificateCallback);
```

Comment:
Is this handler is global, correct? (As opposed to per IPC stream.) The current `subscribeToServerCertificateUpdates` code is de-duping based on callback only, so you may need a unique callback here otherwise only a single client can call this API